File: exercism/typescript/raindrops/raindrops.ts
const factorMap = new Map<number,string>([
  [3, 'Pling'],
  [5, 'Plang'],
  [7, 'Plong']
]);

export function convert(num: number): string {
  let raindropSound: string = '';
  for (let factor of factorMap.keys()) {
    raindropSound += num % factor === 0 ? factorMap.get(factor) : '';
  }

  return raindropSound || num.toString();
}

File: exercism/python/raindrops/raindrops.py
factorMap = {
    3: 'Pling',
    5: 'Plang',
    7: 'Plong'
}

def convert(number: int) -> str:
    return ''.join(drop for factor, drop in factorMap.items() if number % factor == 0) or str(number)
